orsopy/fileio/base.py

```diff
@@ -33,8 +44,14 @@ def _clean(self):
         :return: Cleaned dictionary
         :rtype: dict
         """
-        return dict((k, v) for (k, v) in self.__dict__.items()
-                    if (v is not None or k not in self.orso_optionals))
+        d = {}
+        for k, v in self.__dict__.items():
```

Comment:
This will only return any attributes set on the instance and ignore anything from the class and inherited from parents. I would expect that e.g. default arguments would therefore be ignored. I would propose:
`[(k, getattr(self, k)) for k in dir(self) if not k.startswith('_') and not callable(getattr(self, k))]`

This should include all non-hidden attributes of the instance, class and inherited and would even allow special cases of sub-classes that want to overwrite it by using the __dir__ method.

Reply:
A potential issue with using `dir` that it doesn't keep the same order as the attributes are define.

Reply:
Turns out `self.__dir__()` and `dir(self)` return different things (https://stackoverflow.com/questions/9659361/order-of-results-from-pythons-dir-function/9659377)